### PR TITLE
Test the routines for the progress status storage

### PR DESCRIPTION
**Pull request by nolar**
> Issue : #13 

Only the tests and few little edge-case fixes for the routines handling the progress storage.

The progress storage is used by the framework to keep the status of the handlers, their errors or delays, and the multi-handler processing in general.

The progress struct itself is already documented in the `kopf.structs.progress` docstring with a sample, see here: https://github.com/zalando-incubator/kopf/blob/master/kopf/structs/progress.py

### CODE REVIEW

File: tests/test_progress.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    origbody = copy.deepcopy(body)
+    patch = {}
+    store_failure(body=body, patch=patch, handler=handler, exc=Exception("some-error"))
+    assert patch == expected
+    assert body == origbody  # not modified
+
+
+@pytest.mark.parametrize('result, body, expected', [
+
+    # With no result, it updates only tghe progress.
```
Comment (samurang87) [inline tests/test_progress.py]: typo :)

Comment (nolar) [reply tests/test_progress.py]: @samurang87 Fixed.

**Comment by samurang87** [comment]
There is an issue with a dependency. Do we need the testing requirements in requirements.txt at all?

**Comment by nolar** [comment]
@samurang87 `freezegun` is Apache-2.0 licensed, this license is compatible with MIT: https://opensource.zalando.com/docs/resources/licensing/

And yes, the requirements are needed. Otherwise, the CI/CD tests will not run in Travis CI. (The `setup.py` does not include `requirements.txt`, it is vice versa: `requirements.txt` depends on `.` plus all the development/testing tools.).
